### PR TITLE
fix: Show proper version on about modal

### PR DESCRIPTION
**Pull request by toninis**
#### Summary

Change the version we depict on the about modal. 
- Show the proper version regardless the environment
- Always show the build number . No need to keep that a secret. 
- Show `n/a` buildNumber when local development and `dev` as version

Related issue: https://github.com/mattermost/mattermost/issues/24034
#### Ticket Link

https://mattermost.atlassian.net/browse/CLD-6573

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by amyblais — APPROVED** [review]

File: webapp/channels/src/components/about_build_modal/about_build_modal.tsx [hunk 1]
```diff
@@ -170,24 +170,17 @@ export default class AboutBuildModal extends React.PureComponent<Props, State> {
             </ExternalLink>
         );
 
-        // Only show build number if it's a number (so only builds from Jenkins)
```
Comment (lieut-data) [inline webapp/channels/src/components/about_build_modal/about_build_modal.tsx:173]: > only builds from Jenkins

🤣

**Review by lieut-data — APPROVED** [review]

File: webapp/channels/src/components/about_build_modal/about_build_modal.tsx (hunk 1 above)
Comment (toninis) [reply webapp/channels/src/components/about_build_modal/about_build_modal.tsx:173]: That old piece of code that follows you everywhere 😄

**Review by isacikgoz — APPROVED** [review]

**Comment by saturninoabril** [comment]
I'll defer review to @marianunez/ICU.

**Pull request merged**